hal/src/main/native/cpp/proto/ControlDataProto.cpp

```diff
@@ -135,28 +132,36 @@ std::optional<mrc::Joystick> wpi::Protobuf<mrc::Joystick>::Unpack(
     Joystick.Axes.Axes()[i] = Axes[i];
   }
 
-  Joystick.Povs.SetCount(Povs.size());
+  Joystick.Buttons.SetCount(Msg.ButtonCount);
+  Joystick.Buttons.Buttons = Msg.Buttons;
 
+  Joystick.Povs.SetCount(Msg.POVCount);
+  uint32_t PovsStore = Msg.POVs;
   for (size_t i = 0; i < Joystick.Povs.GetCount(); i++) {
-    Joystick.Povs.Povs()[i] = Povs[i];
+    uint8_t Val = PovsStore & 0xF;
+    PovsStore >>= 4;
+    Joystick.Povs.Povs()[i] = Val;
   }
 
-  Joystick.Buttons.SetCount(Msg.ButtonCount);
-  Joystick.Buttons.Buttons = Msg.Buttons;
-
   return Joystick;
 }
 
 bool wpi::Protobuf<mrc::Joystick>::Pack(OutputStream& Stream,
                                         const mrc::Joystick& Value) {
   wpi::PackCallback AxesCb{Value.Axes.Axes()};
-  wpi::PackCallback PovsCb{Value.Povs.Povs()};
+
+  uint32_t PovsStore = 0;
+  for (size_t i = 0; i < Value.Povs.GetCount(); i++) {
```

Comment:
Is Povs[0] the least significant nybble or the most significant nybble? Unpack seems to think least significant, while Pack seems to think most significant.

Reply:
You are correct. Fixed the bug on the other end (This end never packs, so I won't reingest).

Reply:
Seems to have been fixed in #7980, good to be resolved. (I can't mark this conversation as resolved myself because it's limited to people with write access and PR authors)